IdentifiabilityODE.mpl

```diff
@@ -557,3 +589,139 @@ CompareDiffVar := proc(dvl, dvr, var_list)
   end if:
   return StringTools[Compare](vr, vl):
 end proc:
+
+
+#===============================================================================
+# Weights
+#===============================================================================
+
+# get rid of unerscore, e.g. x1(t) -> x1_ -> x1
+get_function_name := f -> parse(convert(FunctionToVariable(f), string)[..-2]):
+
+# check if depends on (t)
+is_function:= f->StringTools[Has](convert(f, string), "(t)"):
+idtfm := x->x;
+
+# check if is derivative
+is_diff := f->type(int(f, t), function(name)):
+
+lhs_name := ff -> if convert(ff, string)[-1] = "_" then parse(convert(ff, string)[..-2]) else ff; end if:
+
+#===============================================================================
+GetStateName := proc(state, x_vars, mu)
+#===============================================================================
+  local state_;
+  if state in mu then
+    return state;
+  end if;
+  state_ := parse(cat(StringTools[Join](StringTools[Split](convert(state, string), "_")[..-2], "_"), "_")):
+  if state_ in x_vars then
+    return state_;
+  end if:
+end proc:
+
+#===============================================================================
+GetMinLevelBFS := proc(s, m, x_vars, y_vars, mu, x_eqs, y_eqs_in, all_vars)

```

Comment:
The names `s` and `m` do not give too much information. Could you either rename them or write a comment about their meaning?